libs/pageserver_api/src/models.rs

```diff
@@ -336,14 +336,25 @@ impl TimelineCreateRequest {
 
 #[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Eq)]
 pub enum ShardImportStatus {
-    InProgress,
+    InProgress(Option<ShardImportProgress>),
     Done,
     Error(String),
 }
+
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Eq)]
+pub struct ShardImportProgress {
```

Comment:
Please provide a means for format evolution. E.g.


```suggestion
#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Eq)]
pub enum ShardImportProgress {
    V1(ShardImportProgressV1),
}
```

Reply:
Not a big fan of this type of format evolution. We can add new fields via serde, but can do the explicit versioning.